youtube_dl/utils.py

```diff
@@ -842,6 +854,41 @@ def http_response(self, req, resp):
     https_response = http_response
 
 
+def make_socks_conn_class(base_class, socks_proxy):
+    assert issubclass(base_class, (
+        compat_http_client.HTTPConnection, compat_http_client.HTTPSConnection))
+
+    url_components = compat_urlparse.urlparse(socks_proxy)
```

Comment:
I guess it's fine. We already do something similar with `compat_urllib_parse._asciire`.